commit 92226e25192a69839b9ae3b66ea7173732c639b3
Author: Hodorgasm <nsane457@gmail.com>
Date:   Thu May 12 19:25:46 2016 -0400

    GCC-6 only makes 'isnan' available in 'std' namespace, and iostream classes are no longer implicitly convertible to void* so use the equivalent '(fs.fail())' instead of '(fs == NULL)'

--- a/src/libprojectM/Common.hpp
+++ b/src/libprojectM/Common.hpp
@@ -63,8 +63,7 @@
 
 #ifdef LINUX
 #include <cstdlib>
-#define projectM_isnan isnan
-
+#define projectM_isnan std::isnan
 #endif
 
 #ifdef WIN32
--- a/src/libprojectM/MilkdropPresetFactory/Parser.cpp
+++ b/src/libprojectM/MilkdropPresetFactory/Parser.cpp
@@ -1406,7 +1406,7 @@
   PerFrameEqn * per_frame_eqn;
   GenExpr * gen_expr;
 
-  if (fs == NULL)
+  if (fs.fail())
     return NULL;
   if (param_string == NULL)
     return NULL;
@@ -1561,7 +1561,7 @@
 
   if (preset == NULL)
     return NULL;
-  if (fs == NULL)
+  if (fs.fail())
     return NULL;
 
   if ((token = parseToken(fs, name)) != tEq)
@@ -1875,7 +1875,7 @@
   /* Null argument checks */
   if (preset == NULL)
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (fs.fail())
     return PROJECTM_FAILURE;
   if (token == NULL)
     return PROJECTM_FAILURE;
@@ -2166,7 +2166,7 @@
 
   if (token == NULL)
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (fs.fail())
     return PROJECTM_FAILURE;
   if (preset == NULL)
     return PROJECTM_FAILURE;
@@ -2348,7 +2348,7 @@
   if (token == NULL)
 
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (fs.fail())
     return PROJECTM_FAILURE;
   if (preset == NULL)
     return PROJECTM_FAILURE;
